src/operator/nn/mkldnn/mkldnn_base.cc

```diff
@@ -304,18 +383,26 @@ void FallBackCompute(FCompute fn, const nnvm::NodeAttrs &attrs,
   MKLDNNStream::Get()->Submit();
 
   std::vector<TBlob> out_blobs(outputs.size());
+  std::vector<NDArray> temp_src, temp_dst;
   for (size_t i = 0; i < out_blobs.size(); i++) {
     NDArray output = outputs[i];
     // ensure output does not use mkldnn mem.
     // for inplace, we already converted & copied input above.
-    if ((req[i] == kWriteTo) || (req[i] == kWriteInplace))
+    if ((req[i] == kWriteTo) || (req[i] == kWriteInplace)) {
       const_cast<NDArray &>(output).InvalidateMKLDNNData();
-    else if (req[i] == kAddTo)
-      output = outputs[i].Reorder2Default();
+    } else if (req[0] == kAddTo && output.IsMKLDNNData()) {
```

Comment:
I wonder if it's necessary that in the case of fallback, the output is still an MKLDNN array?
I know the unit test can trigger the problem.

Reply:
if the output is not MKLDNN then we do not need to enter this block because we will able to call the FCompute function on the output directly.